### DISCUSSION TITLE
Importing Old Pictures - Asset Upgraded but can't find the original asset ?

### DISCUSSION THREAD

**Discussion by R-Nabil** [simulot/immich-go#474] [General]
Hello,

I've been trying to use Immich-Go (dry-run mode) with old pictures I have to figure out whether these were already on my immich or not. I'm getting several messages like :

 server's asset upgraded with the input | file=IMAG0308.jpg reason="An asset with the same name:\"IMAG0308.jpg\" and date:\"2011-02-08 19:11:29\" but with smaller size:182.6 KB exists on the server. Replace it." time="2024-09-06T08:13.09 UTC"

But if I go on the server, I do not find any asset named like this, or any asset on that date in general.

Am I missing something ?

Thanks

**Comment by simulot** [comment]
The name `IMAG0308.jpg` comes from immich. (not the physical file name you may find on the server's disk)

**Reply by R-Nabil** [reply]
Still. There is no asset for that mentioned date at all on Immich. Like really none at all

**Reply by simulot** [reply]
A side effect of the `-dry-run` option? Maybe the file is duplicated in your input, and when immich-go finds it a second time, it believes it has already uploaded it.

**Comment by R-Nabil** [comment]
Ah. Interesting hypothesis. It is indeed likely I have 2 copies of the same asset, one that was downsized by Google most likely, and the original one.

Would you have a suggestion on how to test this out ? I could always write a short script to go through all my assets and find the ones for that given date.

Also, assuming this is the case, what does the total number of asset uploaded means ? Say I have only these 2 duplicates. It uploads the lower quality first then upgrades it with the higher quality. The total upload would be 2 even though Immich would only have 1 extra asset right ?

Thank you for your time and help

**Reply by simulot** [reply]
How it's possible to  have the original and the downgraded asset in the same takeout? I haven't explicitly tested this case. 
Or maybe, you are not importing a takeout archive.

Anyway, the -dry-out behavior looks good: the 1st encountered file is uploaded, than when the best quality version is found, immich-go should replace the downgraded quality image of the server by the better one.

**Reply by R-Nabil** [reply]
It's just a mess of old pictures inconsistently kept in various different folder merged and un merged constantly. Just a mess. 

I'm pretty sure most of them are already on Immich. Problem is there is quite a bit of them that are not mine and I'm trying to find a programmatic/systematic way to find out about this.

**Reply by simulot** [reply]
Immich-go  cares of duplicates files having same name, and same date of capture.
Immich cares about true binary duplicates, eg files having the same content, byte by byte.
And at the last resort, Immich has introduced a dedup functionality.

You can also look to another community projects:
https://main.preview.immich.app/docs/community-projects
